### PR TITLE
multifile support: guile compile.scm a.ml b.ml c.ml -o foo.byte

### PR DESCRIPTION
**Pull request by gasche**
The declarations of each file are placed into a module whose name is
derived from the filename in the standard OCaml way.

In particular, this changes slightly the semantics of single-file
invocations: now foo.ml is interpreted as if its declarations were
wrapped into a Foo module. The observable behavior of program does not
change, of course, as we only observe the side-effects of the
evaluation.

(This should not change the compilation either, as structure items,
even nested, are compiled into globals. I checked that the generated
bytecode programs for hello.ml keeps the same size.)

### CODE REVIEW

File: miniml/compiler/compile.scm [hunk 1]
```diff
@@ -46,7 +47,14 @@
...
 
-(define ml-parser
+; note: ml-parser takes a dummy argument, because we were not able
+; to call the same lalr-parser on several input files in a reliable way.
+; (There is no documentation about this, or in general very little
+; documentation about lalr-parser.) The bug we would observe
+; is that calling (ml-parser lexer errorp) a second time after changing
+; the default input port would return an empty document. There seems
+; to be some per-parser global state that we don't know how to (re)initialize.
+(define (ml-parser _)
```
Comment (Ekdohibs) [inline miniml/compiler/compile.scm:57]: I think you could use a no-argument function here: `(define (ml-parser) ...)`.
Also, I'm seeing a parser called several times in the documentation [here](http://wiki.call-cc.org/eggref/5/lalr), but there is probably a difference with what we are doing.

**Review by Ekdohibs — COMMENTED** [review]
Seems good! However, I'd say we need some way to have a module that is already open at the beginning, so we can define a standard library for primitives such as `(+)` etc.

File: miniml/compiler/compile.scm (hunk 1 above)
Comment (gasche) [reply miniml/compiler/compile.scm]: Good idea, thanks, I made the change and rebased.

**Comment by gasche** [comment]
I'm not sure about when we would need support for `-open Foo`, and would rather keep it outside the scope of the present PR -- I have other changes to the testsuite that depend on the feature.

**Comment by Ekdohibs** [comment]
The main reason was that previously, the script that assembled all `.ml`s into a single file did put one file at the beginning outside of a `struct ... end` so that it was accessible to all other files. However, we can do that later.

**Pull request merged**